src/main/kotlin/com/example/petshopkotlin/pet/PetService.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.petshopkotlin.pet
+
+import com.example.petshopkotlin.pet.model.Pet
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Service
+
+@Service
+class PetService(val repo: PetRepository) {
+    fun getPets() = repo.findAllByOrderByIdDesc()
+
+    fun addPet(pet: Pet): Pet {
+        val errors = validatePet(pet)
+        require(errors.isBlank()) { errors }
```

Comment:
I would move the error throwing logic into the validate method, since it's tightly coupled anyway. However, it would make the validate method have a side effect. Another, more functional approach, would be to make the method return an exception?-type, and throw it here if it is present.